### PR TITLE
[CIRC-439] Send notice on manual due date change

### PR DESCRIPTION
**Pull request by dmytrotur**
## Purpose
Add sending panton notice on manual due date change
Resolves: [CIRC-439](https://issues.folio.org/browse/CIRC-439)

## Approach
- Add a new test to `ChangeDueDateTests` that verifes sending **manual due date change** notice
- Move code for sending loan notice from `CheckOutByBarcodeResource` and `RenewalResource` to `LoanNoticeSender`
- Use `LoanNoticeSender` on loan update
- Add fetching **location**, **material type**, **loan type**, and **loan policy** on loan update

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java [hunk 1]
```diff
@@ -92,6 +92,9 @@ public static JsonObject createCheckInContext(CheckInProcessRecords records) {
   }
 
   public static JsonObject createUserContext(User user) {
+    if (user == null) {
```
Comment (mtraneis) [inline src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java]: Can the user ever be null? If so, in what context and how does the caller of this method handle it?

File: src/main/java/org/folio/circulation/resources/LoanCollectionResource.java [hunk 2]
```diff
@@ -128,7 +130,7 @@ void replace(RoutingContext routingContext) {
     final Clients clients = Clients.create(context, client);
     final RequestQueueRepository requestQueueRepository = RequestQueueRepository.using(clients);
     final ServicePointRepository servicePointRepository = new ServicePointRepository(clients);
-    final ItemRepository itemRepository = new ItemRepository(clients, false, false, false);
+    final ItemRepository itemRepository = new ItemRepository(clients, true, true, true);
```
Comment (mtraneis) [inline src/main/java/org/folio/circulation/resources/LoanCollectionResource.java]: Can you explain why we need to fetch location, material type, and loan type?

File: src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java [hunk 3]
```diff
@@ -26,7 +30,27 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
...
+      .withTiming(NoticeTiming.UPON_AT)
+      .withNoticeContext(noticeContext)
+      .build();
+
+    acceptNoticeEvent(noticeEvent);
+    return succeeded(records);
+  }
+
   public void acceptNoticeEvent(PatronNoticeEvent event) {
+    if (event.getUser() == null || event.getItem() == null) {
```
[the remark this answers is not in the document]
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java]: Removed this check as unnecessary (now the condition is verified before this method).

File: src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java (hunk 1 above)
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/notice/TemplateContextUtil.java]: Removed this check as unnecessary (now the condition is verified before this method).

File: src/main/java/org/folio/circulation/resources/LoanCollectionResource.java (hunk 2 above)
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/resources/LoanCollectionResource.java]: This information is needed for **patron notice template** context. We have the ability to insert **location**, **material type**, and **loan type** the information into patron notices.

File: src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java [hunk 4]
```diff
@@ -26,7 +30,27 @@ public PatronNoticeService(PatronNoticePolicyRepository noticePolicyRepository,
     this.patronNoticeClient = clients.patronNoticeClient();
   }
 
+  public Result<LoanAndRelatedRecords> sendLoanNotice(LoanAndRelatedRecords records, NoticeEventType eventType) {
```
[the remark this answers is not in the document]
Comment (mtraneis) [reply src/main/java/org/folio/circulation/domain/notice/PatronNoticeService.java]: Thanks. I like that LoanNoticeSender was created.

**Pull request merged**